rustls/src/client/ech.rs

```diff
@@ -26,6 +27,32 @@ use crate::{
     PeerIncompatible, PeerMisbehaved, ProtocolVersion, Tls13CipherSuite,
 };
 
+/// Controls how Encrypted Client Hello (ECH) is used in a client handshake.
+#[derive(Clone, Debug)]
+pub enum EchMode {
+    /// ECH is enabled and the ClientHello will be encrypted based on the provided
+    /// configuration.
+    Enable(EchConfig),
+
+    /// No ECH configuration is available but the client should act as though it were.
+    ///
+    /// This is an anti-ossification measure, sometimes referred to as "GREASE"[^0].
+    /// [^0]: <https://www.rfc-editor.org/rfc/rfc8701>
+    Placeholder(EchPlaceholderConfig),
```

Comment:
Note: I've deliberately avoided using "GREASE" in the user-facing names for this. I think it's much too in-the-weeds to be a useful term for a public API. "Placeholder" seemed better, but I'm open to bikeshedding.

Reply:
Totally doesn't matter. I know more about TLS than most, but when I saw "GREASE", I thought "oh what is this?" Maybe with some more colorful language.

I think you can get some good test material from the draft-07 code. You'll have to update it, but basically you add the "hex" crate as a dev dependency and then you can check every phase of the ECH code, but keep it in ASCII.

Reply:
> Note: I've deliberately avoided using "GREASE" in the user-facing names for this. I think it's much too in-the-weeds to be a useful term for a public API. "Placeholder" seemed better, but I'm open to bikeshedding.

I feel like chances of `Grease` getting recognized as what it is are better than `Placeholder` getting recognized as the mechanism for grease? So I'd prefer to stick to the more familiar name.

Reply:
Really doesn't matter, but GREASE is ok too. It's an acronym, so it should be all-caps:

https://www.rfc-editor.org/rfc/rfc8701.html

"Generate Random Extensions And Sustain Extensibility"

But note that had to capitalize "And", which one not normally do in English. :)

Reply:
Rust naming conventions prescribe that acronyms get title-cased, so I think that spelling is idiomatic.

Reply:
Yeah, I can see your point here. It's just that it is a wrong acronym, so its title-casing is also a wrong thing. But it's probably best to say "Grease" and never discuss it again.

Reply:
>  So I'd prefer to stick to the more familiar name.

I don't feel strongly, I will revert to using `Grease` :+1: